### ISSUE TITLE
Add a Pkg.audit command to check dependencies against reported vulnerabilities

### ISSUE THREAD

**Issue by SaschaMann** [JuliaLang/Pkg.jl#1975]
**TL;DR:** A new command `Pkg.audit()` that checks installed dependencies/environment files for reported security vulnerabilities; inspired by [`npm audit`](https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities).

This is meant to start a discussion first, rather than providing details on how to implement the feature.

---

`npm` has a command to check dependencies against security vulnerabilities known to the registry. Imo it would be useful to have a similar command in Julia’s package manager. This primarily benefits (deployed) Julia applications, not libraries.

Updating all dependencies, while recommended, often comes with some work to ensure that everything will still work after the update, and in more secure environments it also requires reviewing the new versions of the dependencies. It’s often common that dependencies of applications are only updated when absolutely necessary. The audit command could be run on a very frequent schedule, so that potential vulnerabilities can be found quicker than the usual update cycles.

`npm` also runs this automatically whenever you install or update any dependency. Further, they provide an `npm audit fix` command that bumps dependency versions if a later version of the package does not contain the vulnerability anymore.

Instead of repeating everything, take a look at https://docs.npmjs.com/auditing-package-dependencies-for-security-vulnerabilities to find out more about the details, what info is shown etc.

---

Do you think this would generally be a useful addition to Pkg?

---

A first step to implement this in Pkg would be a way for package authors to mark versions as potentially affected by a vulnerability, as well as attaching some information on the details of the vulnerability (even if it's just a link to an issue).

Ideally you’d also want to allow others to report them, but that would likely require human work to verify the reports. Publishing more info on the vulnerabilities, e.g. like the [security advisories](https://www.npmjs.com/advisories) on npmjs.com, would also be nice in the future, but is not needed for basic functionality.
